src/main/scala/chisel3/simulator/SimulatorAPI.scala

```diff
@@ -74,12 +74,12 @@ trait SimulatorAPI {
     */
   def simulate[T <: Module](
     module:                => T,
-    layerControl:          LayerControl.Type = LayerControl.EnableAll,
+    chiselSettings:        ChiselSettings[T] = ChiselSettings.default[T],
```

Comment:
Note: because the `default` factory is necessarily given a type parameter with a lower bound (`A <: RawModule]`), this necessitates giving it a type parameter on line 77 here. If I don't do this, this will infer to `Nothing` and everything blows up.

There may be a way to untangle this.